core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -1206,6 +1205,50 @@ private void unparseOtherFunction(SqlWriter writer, SqlCall call, int leftPrec,
     }
   }
 
+  private void unParseRegexpLike(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+    SqlWriter.Frame ifFrame = writer.startFunCall("IF");
+    unParseRegexpContains(writer, call, leftPrec, rightPrec);
+    writer.sep(",");
+    writer.literal("1");
+    writer.sep(",");
+    writer.literal("0");
+    writer.endFunCall(ifFrame);
+  }
+
+  private void unParseRegexpContains(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+    SqlWriter.Frame regexContainsFrame = writer.startFunCall("REGEXP_CONTAINS");
+    call.operand(0).unparse(writer, leftPrec, rightPrec);
+    writer.print(", r");
+    unParseRegexString(writer, call, leftPrec, rightPrec);
+    writer.endFunCall(regexContainsFrame);
+  }
+
+  private void unParseRegexString(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+    if (call.getOperandList().size() == 3) {
+      SqlCharStringLiteral modifiedRegexString = getModifiedRegexString(call);
+      modifiedRegexString.unparse(writer, leftPrec, rightPrec);
+    } else {
+      call.operand(1).unparse(writer, leftPrec, rightPrec);
+    }
+  }
+
+  private SqlCharStringLiteral getModifiedRegexString(SqlCall call) {
+    String matchArgument = call.operand(2).toString().replaceAll("'", "");
+    switch (matchArgument) {
+    case "i":
+      String caseInsensitiveLiteral = "(?i)";
+      String updatedRegexForI = caseInsensitiveLiteral.concat(call.operand(1).toString()
+          .replaceAll("^'|'$", ""));
+      return SqlLiteral.createCharString(updatedRegexForI, SqlParserPos.ZERO);
+    case "x":
+      String updatedRegexForX = call.operand(1).toString().replaceAll("\\s+", "")
+          .replaceAll("^'|'$", "");
+      return SqlLiteral.createCharString(updatedRegexForX, SqlParserPos.ZERO);
+    default:
```

Comment:
logically wrong thing to do. we are making a decision on the basis of the 3rd operand & are modifying the 3rd operand so the default should not return the 2nd operand. If you always have to unparse the 2nd operand, you can do that in the unParseRegexString method by removing the else block

Reply:
In the case of the REGEXP_LIKE function with 2 arguments-
 While accessing the 3rd argument of REGEXP_LIKE function ( call.operand(2) ), it throws ArrayIndexOutOfBound Exception. So to handle this scenario, we have added that else block.